src/pages/client/pages/ClientAddTunnelPage/components/AddTunnelFormCard/AddTunnelFormCard.tsx

```diff
@@ -0,0 +1,279 @@
+import { useEffect, useMemo, useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { useI18nContext } from '../../../../../../i18n/i18n-react';
+import { FormInput } from '../../../../../../shared/defguard-ui/components/Form/FormInput/FormInput';
+import { ArrowSingle } from '../../../../../../shared/defguard-ui/components/icons/ArrowSingle/ArrowSingle';
+import {
+  ArrowSingleDirection,
+  ArrowSingleSize,
+} from '../../../../../../shared/defguard-ui/components/icons/ArrowSingle/types';
+import { Button } from '../../../../../../shared/defguard-ui/components/Layout/Button/Button';
+import {
+  ButtonSize,
+  ButtonStyleVariant,
+} from '../../../../../../shared/defguard-ui/components/Layout/Button/types';
+import { Card } from '../../../../../../shared/defguard-ui/components/Layout/Card/Card';
+import { Helper } from '../../../../../../shared/defguard-ui/components/Layout/Helper/Helper';
+import { useToaster } from '../../../../../../shared/defguard-ui/hooks/toasts/useToaster';
+import { generateWGKeys } from '../../../../../../shared/utils/generateWGKeys';
+import { clientApi } from '../../../../clientAPI/clientApi';
+
+type FormFields = {
+  name: string;
+  pubkey: string;
+  prvkey: string;
+  address: string;
+  server_pubkey: string;
+  allowed_ips?: string;
+  endpoint: string;
+  dns?: string;
+  persistent_keep_alive: number;
+  route_all_traffic: boolean;
+  pre_up?: string;
+  post_up?: string;
+  pre_down?: string;
+  post_down?: string;
+};
+const defaultValues: FormFields = {
+  name: '',
+  pubkey: '',
+  prvkey: '',
+  address: '',
+  server_pubkey: '',
+  allowed_ips: '',
+  endpoint: '',
+  dns: '',
+  persistent_keep_alive: 25, // Adjust as needed
+  route_all_traffic: false,
+  pre_up: '',
+  post_up: '',
+  pre_down: '',
+  post_down: '',
+};
+
+export const AddTunnelFormCard = () => {
+  const { LL } = useI18nContext();
+  const { parseConfig, saveTunnel } = clientApi;
+  const toaster = useToaster();
+
+  const localLL = LL.pages.client.pages.addTunnelPage.forms.initTunnel;
+  /* eslint-disable no-useless-escape */
+  const schema = useMemo(
+    () =>
+      z.object({
+        name: z.string().trim().min(1, LL.form.errors.required()),
+        pubkey: z.string().trim().min(1, LL.form.errors.required()),
+        prvkey: z.string().trim().min(1, LL.form.errors.required()),
+        server_pubkey: z.string().trim().min(1, LL.form.errors.required()),
+        address: z.string().refine((value) => {
+          // Regular expression to match IPv4 CIDR and IPv6 CIDR
+          const cidrRegex =
+            /^(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\/\d{1,2}|[0-9a-fA-F:.]+\/\d{1,3})$/;
+          return cidrRegex.test(value);
+        }, LL.form.errors.invalid()),
+        endpoint: z.string().refine((value) => {
+          // Regular expression to match IPv4, IPv6, or domain name with port
+          const endpointRegex = /^([0-9a-fA-F:\.]+|\w+(\.\w+)+)(:\d+)?$/;
```

Comment:
Same as above and also make sure to accept localhost bcs users already posted issues about it. Assuming this will be used by more informed users, validating endpoints could be considered as not needed.